pkg/sdk/testint/security_integrations_gen_integration_test.go

```diff
@@ -0,0 +1,380 @@
+		assert.Contains(t, details, sdk.SecurityIntegrationProperty{Name: "SAML2_SNOWFLAKE_ISSUER_URL", Type: "String", Value: d.snowflakeIssuerURL, Default: ""})
+		assert.Contains(t, details, sdk.SecurityIntegrationProperty{Name: "SAML2_SNOWFLAKE_ACS_URL", Type: "String", Value: d.snowflakeAcsURL, Default: ""})
+		assert.Contains(t, details, sdk.SecurityIntegrationProperty{Name: "ALLOWED_USER_DOMAINS", Type: "List", Value: d.allowedUserDomains, Default: "[]"})
+		assert.Contains(t, details, sdk.SecurityIntegrationProperty{Name: "ALLOWED_EMAIL_PATTERNS", Type: "List", Value: d.allowedEmailPatterns, Default: "[]"})
+	}
+
+	t.Run("CreateSAML2", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		createSAML2Integration(t, id, func(r *sdk.CreateSAML2SecurityIntegrationRequest) {
+			r.WithAllowedEmailPatterns([]sdk.EmailPattern{{Pattern: "^(.+dev)@example.com$"}}).
+				WithAllowedUserDomains([]sdk.UserDomain{{Domain: "example.com"}}).
+				WithComment(sdk.Pointer("a")).
+				WithSaml2EnableSpInitiated(sdk.Pointer(true)).
+				WithSaml2ForceAuthn(sdk.Pointer(true)).
+				WithSaml2PostLogoutRedirectUrl(sdk.Pointer("http://example.com/logout")).
+				WithSaml2RequestedNameidFormat(sdk.Pointer("urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified")).
+				WithSaml2SignRequest(sdk.Pointer(true)).
+				WithSaml2SnowflakeAcsUrl(&acsURL).
+				WithSaml2SnowflakeIssuerUrl(&issuerURL).
+				WithSaml2SpInitiatedLoginPageLabel(sdk.Pointer("label"))
+		})
+		details, err := client.SecurityIntegrations.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSAML2Describe(details, saml2details{
+			provider:                  "Custom",
+			enableSPInitiated:         "true",
+			spInitiatedLoginPageLabel: "label",
+			ssoURL:                    "https://example.com",
+			issuer:                    "test",
+			requestedNameIDFormat:     "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+			forceAuthn:                "true",
+			postLogoutRedirectUrl:     "http://example.com/logout",
+			signrequest:               "true",
+			comment:                   "a",
+			snowflakeIssuerURL:        issuerURL,
+			snowflakeAcsURL:           acsURL,
+			allowedUserDomains:        "[example.com]",
+			allowedEmailPatterns:      "[^(.+dev)@example.com$]",
+		})
+
+		si, err := client.SecurityIntegrations.ShowByID(ctx, id)
+		require.NoError(t, err)
+		assertSecurityIntegration(t, si, id, "SAML2", false, "a")
+	})
+
+	t.Run("CreateSCIM", func(t *testing.T) {
+		networkPolicy, networkPolicyCleanup := testClientHelper().NetworkPolicy.CreateNetworkPolicy(t)
+		t.Cleanup(networkPolicyCleanup)
+
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		createSCIMIntegration(t, id, func(r *sdk.CreateSCIMSecurityIntegrationRequest) {
+			r.WithComment(sdk.Pointer("a")).
+				WithNetworkPolicy(sdk.Pointer(sdk.NewAccountObjectIdentifier(networkPolicy.Name))).
```

Comment:
Hmm, so it works if that test passes, then maybe the documentation should be adjusted that single quotes are incorrect (to check) or that the double quotes are also fine.

Reply:
Raised this in jira.